File: sort-mapper-example-java/src/main/java/kh/farrukh/sortmapperexamplejava/course/model/CourseShortInfoResponseDTO.java
package kh.farrukh.sortmapperexamplejava.course.model;

import kh.farrukh.sortmapper.annotation.SortField;

public record CourseShortInfoResponseDTO(
    Long id,
    String name,
    @SortField("teacher.firstName")
    String teacher
) {

}

File: sort-mapper-example/src/main/kotlin/kh/farrukh/sortmapperexample/course/model/CourseShortInfoResponseDTO.kt
package kh.farrukh.sortmapperexample.course.model

import kh.farrukh.sortmapper.annotation.SortField

data class CourseShortInfoResponseDTO(
    val id: Long,
    val title: String,
    @SortField("teacher.firstName")
    val teacher: String
)